runtime/src/snapshot_utils.rs

```diff
@@ -970,6 +973,88 @@ pub fn bank_from_snapshot_archives(
     accounts_db_config: Option<AccountsDbConfig>,
     accounts_update_notifier: Option<AccountsUpdateNotifier>,
     exit: &Arc<AtomicBool>,
+) -> Result<(Bank, BankFromArchiveTimings)> {
+    // The first try is to skip unpacking the appendvec files from the snapshot
+    // if they already exist on disk.
```

Comment:
An alternative to this try-fallback method is we just pass a command line option: `--skip-unpack` and go down whichever path and fail hard.

Reply:
With that --skip-unpack flag, we need to handle 4 scenarios: --skip-unpack on/off, the file existing yes/no.  what should be the action when --skip-unpack is on, but the file does not exist, or when --skip-unpack is off, but the file does exist?

I think at the upper level deleting the files or not already convey the info we want to skip-unpack or not.  So the function can handle this  in the simpler "if not existing then unpack" way.

If we do use that flag, then the upper level deleting action can be removed, and deleting is done if --skip-unpack is off but the file exists.  That is effectively the same.  I am fine either way.

Reply:
try-fallback method is needed, because if the files exist and used, but bank.verify_snapshot_hash fails, we still want to delete the files and retry with the unpacking done first.

Reply:
> try-fallback method is needed, because if the files exist and used, but bank.verify_snapshot_hash fails, we still want to delete the files and retry with the unpacking done first.

I guess I'm unsure if this is the behavior we actually want. If I was expecting ledger-tool to run using my existing append-vecs, and the bank verification fails, it might be useful to not delete the files and investigate.

I don't have that strong feelings on it, so maybe we just keep it as is (try-fallback) and if the need arises we can change it.

Reply:
Looked into adding akip-unpack flag into unpack_archive, and adding another "for entry in archive.entries()? {" loop to check the file existence ahead of time.  There is quite much other checking in the original loop.  not clear to me how much should be replicated into this additional check loop.  I think the chance of accounts files partially missing is small and the benefit of saving unpacking time in the partial file missing case is limited.  It is probably better to keep the current log as is, because it is simpler and clearer.